### PR TITLE
Issue/v4/1067 custom plugins

### PR DESCRIPTION
**Pull request by stefanorosanelli**
This PR solves #1067 

All custom/3rd party plugins are loaded via `Plugins::loadFromConfig()` reading from `Plugins` configuration key

in `config/app.php` put a `Plugins` key like in `config/app.default.php`

### CODE REVIEW

File: plugins/BEdita/Core/src/Utility/Plugins.php [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+namespace BEdita\Core\Utility;
+
+use Cake\Core\Configure;
+use Cake\Core\Plugin;
+
+/**
+ * Plugins utility class
+ *
+ */
+class Plugins
```
Comment (fquffio) [inline plugins/BEdita/Core/src/Utility/Plugins.php]: I think that having `Plugin` and `Plugins` might lead to some confusion. I'd rather rename this class to `\BEdita\Core\Plugin` and let it extend `\Cake\Core\Plugin`. We could then use `\BEdita\Core\Plugin` everywhere with the added benefits of `loadFromConfig()` method.

File: config/app.default.php [hunk 2]
```diff
@@ -370,5 +370,21 @@
...
+     *
+     * - `debugOnly` - boolean - (default: false) Whether or not you want to load the plugin when in 'debug' mode only
+     * - `bootstrap` - boolean - (default: false) Whether or not you want the $plugin/config/bootstrap.php file loaded.
+     * - `routes` - boolean - (default: false) Whether or not you want to load the $plugin/config/routes.php file.
+     * - `ignoreMissing` - boolean - (default: false) Set to true to ignore missing bootstrap/routes files.
+     * - `autoload` - boolean - (default: false) Whether or not you want an autoloader registered
+     */
+    'Plugins' => [
+        'DebugKit' => ['debugOnly' => true, 'bootstrap' => true],
+        'Migrations' => ['debugOnly' => true],
```
Comment (fquffio) [inline config/app.default.php]: **Migrations** should be loaded even if debug mode is turned off. It is required in order to run `bin/cake migrations migrate` on production systems when some update to the database schema is available.

Instead, it should probably be loaded only in CLI environments. Looking at [CakePHP skeleton app](https://github.com/cakephp/app/blob/master/config/bootstrap_cli.php), both **Migrations** and **Bake** are loaded in `config/bootstrap_cli.php`.

Comment (fquffio) [reply config/app.default.php]: Another option is to build a wrapper shell around Migration's default shell, and load the plugin only when strictly needed… but loading it in all CLI environments would probably be simpler.

Comment (stefanorosanelli) [reply config/app.default.php]: Agreed, we may load *Migrations* and *Bake* explicitly in `config/bootstrap_cli.php` only and remove *Migrations* from app default configuration.

File: plugins/BEdita/Core/src/Utility/Plugins.php (hunk 1 above)
Comment (stefanorosanelli) [reply plugins/BEdita/Core/src/Utility/Plugins.php]: Done

File: config/app.default.php (hunk 2 above)
Comment (stefanorosanelli) [reply config/app.default.php]: Done

**Review by fquffio — APPROVED** [review]

**Pull request merged**